nncf/common/pruning/operations.py

```diff
@@ -311,6 +313,22 @@ class ReshapePruningOp(BasePruningOp):
     def _is_flatten(node: NNCFNode) -> bool:
         return len(node.layer_attributes.output_shape) == 2
 
+    @staticmethod
+    def _is_not_mixing_dim(node: NNCFNode, graph) -> bool:
+        input_edges = graph.get_input_edges(node)
+        input_shape = input_edges[0].tensor_shape
+
+        output_edges = graph.get_output_edges(node)
+        if not output_edges:
+            return True
+        output_shape = output_edges[0].tensor_shape
+
+        if len(input_shape)==len(output_shape) and set(input_shape)==set(output_shape):
+            input_shape_idx = np.argsort(input_shape)
+            output_shape_idx = np.argsort(output_shape)
+            return np.allclose(input_shape_idx, output_shape_idx)
```

Comment:
Case (a, b) -> (b, a, 1) will return True here. 

Please implement check by dims alignment like this:

Positive case
![image](https://user-images.githubusercontent.com/74656388/186870521-7aaff059-bf82-4cf6-b69a-1292fe386c12.png)

Negative case
![image](https://user-images.githubusercontent.com/74656388/186870808-c4a1e54e-d85a-444e-8767-058d2d0db547.png)

Reply:
To check all of the case in here is so complicated. Also, there are not so much case with split. I do not understand we need to cover all case in here which is not realistic.

Reply:
I have no idea to cover (a, b) -> (b, a, 1). And is there any real case to do like this?

Reply:
As you mean, the case of (1, a, b) -> (1, b, a) is negative case. I already added the case of Split->Reshape model. It could cover (1, 8, 4, 8) -> (1, 4, 8, 8). If you mean I need to cover more case like (a, b*c, d, e) -> (a, b, c, d*e), I just thought the case of the below.
![SplitReshapePositive](https://user-images.githubusercontent.com/91641705/187597830-631e69ff-5599-4aa4-989b-5af20f0d0b66.png)
I'd appreciate it if you could give me your opinion on this model.